### PR TITLE
stateful dlines

### PR DESCRIPTION
**Pull request by mberezhna**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,32 @@
...
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete newState[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
```
Comment (Fluebubble) [inline src/transformStateWithClones.js:28]: Add `default` after all cases with throwing `Error`. [Read ](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) (third paragraph).

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,32 @@
...
+    }
+
+    stateClones.push(newState);
+  }
+  stateClones.shift();
```
Comment (Fluebubble) [inline src/transformStateWithClones.js:33]: You can use `...spread` instead of `Object.assign` for easier reading and not use `shift()` on every iteration

**Review by Fluebubble — CHANGES_REQUESTED** [review]
Great work!😉 Needs a small edit

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,38 @@
...
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateCopy[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      default: Error(`Unexpected action type: ${action.type}`);
```
Comment (polosanya) [inline src/transformStateWithClones.js:34]:
```suggestion
      default: throw new Error(`Unexpected action type: ${action.type}`);
```

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,38 @@
...
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    let stateCopy = { ...cloneState };
+
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...action.extraData,
```
Comment (polosanya) [inline src/transformStateWithClones.js:20]: You can use Object.assign instead

**Review by polosanya — APPROVED** [review]
Good job!

**Pull request closed without merging**